reactive/kubernetes_master.py

```diff
@@ -1361,30 +1367,30 @@ def remove_rbac_resources():
 @when('nrpe-external-master.available')
 @when_any('config.changed.nagios_context',
           'config.changed.nagios_servicegroups')
-def update_nrpe_config(unused=None):
-    services = (
-        'snap.kube-apiserver.daemon',
-        'snap.kube-controller-manager.daemon',
-        'snap.kube-scheduler.daemon'
-    )
+def update_nrpe_config():
+    services = ['snap.{}.daemon'.format(s) for s in master_services]
+
+    plugin = install_nagios_plugin_from_file('templates/nagios_plugin.py',
+                                             'check_k8s_master.py')
     hostname = nrpe.get_nagios_hostname()
     current_unit = nrpe.get_nagios_unit_name()
     nrpe_setup = nrpe.NRPE(hostname=hostname)
     nrpe.add_init_service_checks(nrpe_setup, services, current_unit)
+    nrpe_setup.add_check('k8s-api-server',
+                         'Verify that the Kubernetes API server is accessible',
+                         str(plugin))
```

Comment:
can we call this check `k8s-master` instead of `k8s-api-server`?  we're checking 3 services on the same machine here, so it seems weird to umbrella them with `api-server` when we actually care about the c-manager and scheduler too.

Reply:
This specific plugin verifies that all snaps are installed and then verifies that it can talk to the API server. It is a bit of a misnomer to call it just apiserver, but it isn't checking that the other services are up. There are other nagios checks for that.